homeassistant/components/myuplink/sensor.py

```diff
@@ -102,4 +142,8 @@ def __init__(
     def native_value(self) -> StateType:
         """Sensor state value."""
         device_point = self.coordinator.data.points[self.device_id][self.point_id]
+        if self.device_class == SensorDeviceClass.ENUM:
```

Comment:
I'd rather make an entity sensor subclass for enum sensors with its own property for `native_value` and an attribute for `options`, so we get rid of those two checks (the check here and the check in `get_description`). Then it's only one check for the enum device class when iterating the data points.